### PR TITLE
firefly-desktop: init at 1.2.0

### PR DESCRIPTION
**Pull request by WolfangAukang**
###### Motivation for this change

#143855

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/firefly-desktop/default.nix [hunk 1]
```diff
@@ -0,0 +1,35 @@
+{ lib, fetchurl, appimageTools }:
+
+let
+  pname = "firefly-desktop";
+  version = "1.2.0";
+  name = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/firefly-desktop/default.nix:6]:
```suggestion
```

File: pkgs/applications/misc/firefly-desktop/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  src = fetchurl {
+    url = "https://github.com/iotaledger/firefly/releases/download/desktop-${version}/${pname}-${version}.AppImage";
+    sha256 = "f3162efcf0407614fd1351af50e95ef180400b747a5cc6b82bc840828a15548d";
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
+
+in appimageTools.wrapType2 {
+  inherit name src;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/firefly-desktop/default.nix:14]:
```suggestion
  inherit pname version src;
```

File: pkgs/applications/misc/firefly-desktop/default.nix [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+let
+  pname = "firefly-desktop";
+  version = "1.2.0";
+  name = "${pname}-${version}";
+  src = fetchurl {
+    url = "https://github.com/iotaledger/firefly/releases/download/desktop-${version}/${pname}-${version}.AppImage";
+    sha256 = "f3162efcf0407614fd1351af50e95ef180400b747a5cc6b82bc840828a15548d";
+  };
+  appimageContents = appimageTools.extractType2 { inherit name src; };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/firefly-desktop/default.nix:11]:
```suggestion
  appimageContents = appimageTools.extractType2 { inherit pname version src; };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/firefly-desktop/default.nix (hunk 1 above)
Comment (WolfangAukang) [reply pkgs/applications/misc/firefly-desktop/default.nix]: Done

File: pkgs/applications/misc/firefly-desktop/default.nix (hunk 3 above)
Comment (WolfangAukang) [reply pkgs/applications/misc/firefly-desktop/default.nix]: Done

File: pkgs/applications/misc/firefly-desktop/default.nix (hunk 2 above)
Comment (WolfangAukang) [reply pkgs/applications/misc/firefly-desktop/default.nix]: Done

**Review by felschr — CHANGES_REQUESTED** [review]
Works well for me. Perhaps update this to the newest version and we should be good to merge.

**Comment by SuperSandro2000** [comment]
Update can be done in another PR.

**Pull request merged**